tests/integration/test_rest_catalog.py

```diff
@@ -16,34 +16,788 @@
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def example_table_metadata_with_no_location(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_rest(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
 
 
 @pytest.mark.integration
-@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog")])
-def test_namespace_exists(catalog: RestCatalog) -> None:
-    if not catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+@pytest.fixture(scope="function")
+@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up")])
+def test_clean_up(catalog: RestCatalog) -> None:
+    for namespaces_tuple in catalog.list_namespaces():
+        namespace_name = namespaces_tuple[0]
+        if TEST_NAMESPACE_IDENTIFIER[0] in namespace_name:
+            for identifier in catalog.list_tables(namespace_name):
+                catalog.purge_table(identifier)
+            if catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+                catalog.drop_namespace(TEST_NAMESPACE_IDENTIFIER)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("catalog,clean_up", [(pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up"))])
+def test_create_namespace_200(catalog: RestCatalog, clean_up: Any) -> None:
+    catalog.create_namespace(TEST_NAMESPACE_IDENTIFIER)
+    assert TEST_NAMESPACE_IDENTIFIER in catalog.list_namespaces()
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("catalog,clean_up", [(pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up"))])
+def test_create_namespace_if_exists_409(catalog: RestCatalog, clean_up: Any) -> None:
+    catalog.create_namespace(TEST_NAMESPACE_IDENTIFIER)
+    catalog.create_namespace_if_not_exists(TEST_NAMESPACE_IDENTIFIER)
+    assert TEST_NAMESPACE_IDENTIFIER in catalog.list_namespaces()
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("catalog,clean_up", [(pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up"))])
+def test_list_namespaces_200(catalog: RestCatalog, clean_up: Any) -> None:
+    catalog.create_namespace(TEST_NAMESPACE_IDENTIFIER)
+    assert catalog.list_namespaces() == [("default",), TEST_NAMESPACE_IDENTIFIER]
```

Comment:
nit: `TEST_NAMESPACE_IDENTIFIER in catalog.list_namespaces()` so we dont care about the other namespace